tests/cpp/test_multi_matmul_scheduler.cpp

```diff
@@ -0,0 +1,609 @@
+    params.async_gmem_load_operands =
+        params.circular_buffer_options.smem_circular_buffer_stage > 1;
+    params.use_smem_epilogue = smem_epilogue;
+    params.splitk_factor = splitk_factor;
+    params.cta_order = cta_order_col_major
+        ? MatmulParams::TileRasterizationOrder::ColumnMajor
+        : MatmulParams::TileRasterizationOrder::RowMajor;
+    params.grid_swizzle_factor = grid_swizzle_factor;
+  }
+
+  void SetUp() {
+    NVFUSER_TEST_CUDA_ARCH_RANGE_GUARD(7, 5, 9, 0);
+  }
+
+  void TearDown() {
+    if (!IsSkipped()) {
+      compareSchedules();
+    }
+  }
+
+  // Get A and B in shapes [M, K] and [K, N] with allocation domains set.
+  std::pair<TensorView*, TensorView*> getInputTVs() {
+    auto tv0 = makeContigTensor(2, DataType::Half);
+    auto tv1 = makeContigTensor(2, DataType::Half);
+
+    if (a_m_inner) {
+      tv0->setAllocationDomain({tv0->axis(1), tv0->axis(0)}, true);
+    }
+
+    if (b_k_inner) {
+      tv1->setAllocationDomain({tv1->axis(1), tv1->axis(0)}, true);
+    }
+    return {tv0, tv1};
+  }
+
+  // Get A and B in shapes [Batch, M, K] and [Batch, K, N] with allocation
+  // domains set.
+  std::pair<TensorView*, TensorView*> getBatchInputTVs() {
+    auto tv0 = makeContigTensor(3, DataType::Half);
+    auto tv1 = makeContigTensor(3, DataType::Half);
+
+    if (a_m_inner) {
+      tv0->setAllocationDomain(
+          {tv0->axis(0), tv0->axis(2), tv0->axis(1)}, true);
+    }
+
+    if (b_k_inner) {
+      tv1->setAllocationDomain(
+          {tv1->axis(0), tv1->axis(2), tv1->axis(1)}, true);
+    }
+    return {tv0, tv1};
+  }
+
+  // Get A and B in already-broadcasted shapes [M, K, 1] and [1, K, N],
+  // possibly with allocation domains sset
+  std::pair<TensorView*, TensorView*> getBroadcastInputTVs() {
+    auto tv0 = makeContigConcreteTensor({-1, 1, -1}, DataType::Half);
+    auto tv1 = makeContigConcreteTensor({1, -1, -1}, DataType::Half);
+
+    if (a_m_inner) {
+      tv0->setAllocationDomain(
+          {tv0->axis(2), tv0->axis(1), tv0->axis(0)}, true);
+    }
+
+    if (b_k_inner) {
+      tv1->setAllocationDomain(
+          {tv1->axis(2), tv1->axis(1), tv1->axis(0)}, true);
+    }
+    return {tv0, tv1};
+  }
+
+  std::pair<at::Tensor, at::Tensor> getInputTensors(
+      int M,
+      int N,
+      int K,
+      bool a_m_inner,
+      bool b_k_inner,
+      bool broadcasted = false) {
+    const auto options =
+        at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0 /*device*/);
+    auto t0 = a_m_inner ? at::randn({M, K}, options).as_strided({M, K}, {1, M})
+                        : at::randn({M, K}, options);
+    auto t1 = b_k_inner ? at::randn({K, N}, options).as_strided({K, N}, {1, K})
+                        : at::randn({K, N}, options);
+    if (broadcasted) {
+      t0 = t0.unsqueeze(-1);
+      t1 = t1.unsqueeze(0);
+    }
+    return {t0, t1};
+  }
+
+  // Recursively compare scalar values
+  void compareScalars(Val* v_orig, Val* v_new) {
```

Comment:
Can this be simplified as:
```C++
return cloner_->clone(v_orig)->sameAs(v_new);
```
If the above isn't just working, then don't bother investing more time to make it work. This is a test that will throw out in the future, so I don't really care about having a clean and elegant implementation.

Reply:
This would work if we had cloned the _scheduled_ fusion, but here I have cloned the _unscheduled_ fusion then scheduled them both. That means that some of these scalars are new in both the original and the clone. When I clone those over using `cloner_->clone(v_orig)` we wind up with a new _undefined_ scalar, which fails the `sameAs` check if the original had a definition.